packages/sdk/src/oracle/common/types/interfaces.ts

```diff
@@ -0,0 +1,177 @@
+import { TransactionConfirmer } from "../utils";
+import * as services from "../services";
+import * as stateTypes from "./state";
+import * as ethersTypes from "./ethers";
+
+export interface Client {
+  setUser: (params: Partial<User>) => string;
+  clearUser: () => string;
+  setActiveRequest: (params: stateTypes.InputRequest) => string;
+  setActiveRequestByTransaction: (params: services.statemachines.setActiveRequestByTransaction.Params) => string;
+  approveCollateral: () => string;
+  proposePrice: (proposedPriceDecimals: string | number) => string;
+  disputePrice: () => string;
+  settle: () => string;
+  switchOrAddChain: () => string;
+  startInterval: (delayMs: number) => void;
+  stopInterval: () => void;
+}
+
+export type WriteCallback<State, Events> = (write: Write<Events>, state: State) => void;
+export type RawWriteCallback<S> = (state: S) => void;
+export type Emit<S> = (state: S, prev: S) => void;
+
+export interface StoreConfig<State, Oracle, Events> {
+  emit: Emit<State>;
+  state: State;
+  Read: NewRead<State, Oracle, Events>;
+  Write: NewWrite<State, Events>;
+  Has: NewHas<State>;
+  Update: NewUpdate<Store<State, Oracle, Events>>;
+}
+export interface RawStore<S> {
+  write: (cb: RawWriteCallback<S>) => void;
+  writeAsync: (cb: RawWriteCallback<S>) => Promise<void>;
+  read: () => S;
+}
+export interface Store<S, O, E> {
```

Comment:
nit: as a reader, single-letter generics can be confusing, as it's unclear what these types are supposed to represent.